nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/move-controller-service/move-controller-service.component.ts

```diff
@@ -0,0 +1,232 @@
+        }
+        this.loadChildOptions(request.childProcessGroupOptions, request.processGroupEntity, processGroups);
+        this.controllerServiceActionProcessGroups = processGroups;
+
+        const firstEnabled = processGroups.findIndex((pg) => !pg.disabled);
+        if (firstEnabled != -1) {
+            this.moveControllerServiceForm.controls['processGroups'].setValue(processGroups[firstEnabled].value);
+        }
+    }
+
+    loadParentOption(
+        breadcrumb: BreadcrumbEntity,
+        parentControllerServices: ControllerServiceEntity[],
+        processGroups: SelectOption[]
+    ) {
+        if (breadcrumb.parentBreadcrumb != undefined) {
+            const parentBreadcrumb = breadcrumb.parentBreadcrumb;
+            if (parentBreadcrumb.permissions.canRead && parentBreadcrumb.permissions.canWrite) {
+                const option: SelectOption = {
+                    text: parentBreadcrumb.breadcrumb.name + ' (Parent)',
+                    value: parentBreadcrumb.breadcrumb.id
+                };
+
+                let errorMsg = '';
+                const descriptors = this.controllerService.component.descriptors;
+                for (const descriptor in descriptors) {
+                    if (descriptors[descriptor].identifiesControllerService != undefined) {
+                        const controllerId = this.controllerService.component.properties[descriptors[descriptor].name];
+                        if (
+                            controllerId != null &&
+                            !parentControllerServices.some((service) => service.id == controllerId)
+                        ) {
+                            errorMsg += '[' + descriptors[descriptor].name + ']';
+                        }
+                    }
+                }
+
+                if (errorMsg != '') {
+                    option.description =
+                        'The following properties reference controller services that would be out of scope for this ' +
+                        'process group: ' +
+                        errorMsg;
+                    option.disabled = true;
+                } else {
+                    option.disabled = false;
+                }
+
+                processGroups.push(option);
+            }
+        }
+    }
+
+    loadChildOptions(
+        childProcessGroupOptions: SelectOption[],
+        currentProcessGroupEntity: any,
+        processGroups: SelectOption[]
+    ) {
+        const referencingComponents: ControllerServiceReferencingComponentEntity[] =
+            this.controllerService.component.referencingComponents;
+        childProcessGroupOptions.forEach((child: SelectOption) => {
+            const option: SelectOption = {
+                text: child.text,
+                value: child.value
+            };
+
+            const root = this.getProcessGroupById(currentProcessGroupEntity.component, child.value ?? '');
```

Comment:
`getProcessGroupById` will return null for a couple different scenarios. When this happens calling `processGroupContainsComponent` will error with the following. The unhandled error leaves the UI in a weird state because the overlay is active but the dialog isn't shown.

```
TypeError: Cannot read properties of null (reading 'contents')
    at _MoveControllerService.processGroupContainsComponent (move-controller-service.component.ts:184:26)
    at move-controller-service.component.ts:166:27
    at Array.forEach (<anonymous>)
    at _MoveControllerService.loadChildOptions (move-controller-service.component.ts:157:34)
    at new _MoveControllerService (move-controller-service.component.ts:99:14)
    at NodeInjectorFactory.MoveControllerService_Factory [as factory] (move-controller-service.component.ts:231:5)
    at getNodeInjectable (core.mjs:6060:44)
    at createRootComponent (core.mjs:17231:35)
    at ComponentFactory.create (core.mjs:17091:29)
    at ViewContainerRef2.createComponent (core.mjs:17498:47) 
```

Reply:
It appears this was happening when `childProcessGroupOptions` contained more than one option. The recursive logic in `getProcessGroupById` (and other methods with similar structure) isn't correct. When there are multiple children the following code

```
                    for (const pg of root.contents.processGroups) {
                        return this.getProcessGroupById(pg, processGroupId);
                    }
```

will return the result from the first child and not attempt to process any subsequent children.

Reply:
I've fixed the recursion logic.